# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'], format="%Y-%m-%d")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [4]:
# Set the date column as the DataFrame index
df.set_index("date", drop=True, inplace=True)
df

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73
...                 ...   ...   ...
2017-08-19  USC00516128  0.09    71
2017-08-20  USC00516128   NaN    78
2017-08-21  USC00516128  0.56    76
2017-08-22  USC00516128  0.50    76
2017-08-23  USC00516128  0.45    76

[19550 rows x 3 columns]

In [5]:
# Drop the date column

### Compare June and December data across all years 

In [6]:
from scipy import stats

In [9]:
# Filter data for desired months
#june = df["date"].dt.month == 6
#june_month=df(june)
#june_month
df = df.reset_index()
june = df.loc[df["date"].dt.month == 6]
june.set_index("date")
june

index       date      station  prcp  tobs
133      133 2010-06-01  USC00519397  0.00    78
134      134 2010-06-02  USC00519397  0.01    76
135      135 2010-06-03  USC00519397  0.00    78
136      136 2010-06-04  USC00519397  0.00    76
137      137 2010-06-05  USC00519397  0.00    77
...      ...        ...          ...   ...   ...
19492  19492 2017-06-26  USC00516128  0.02    79
19493  19493 2017-06-27  USC00516128  0.10    74
19494  19494 2017-06-28  USC00516128  0.02    74
19495  19495 2017-06-29  USC00516128  0.04    76
19496  19496 2017-06-30  USC00516128  0.20    75

[1700 rows x 5 columns]

In [11]:
# december temperature data
df = df.reset_index()
december = df.loc[df["date"].dt.month == 12]
december.set_index("date")
december


level_0  index       date      station  prcp  tobs
305        305    305 2010-12-01  USC00519397  0.04    76
306        306    306 2010-12-03  USC00519397  0.00    74
307        307    307 2010-12-04  USC00519397  0.00    74
308        308    308 2010-12-06  USC00519397  0.00    64
309        309    309 2010-12-07  USC00519397  0.00    64
...        ...    ...        ...          ...   ...   ...
19323    19323  19323 2016-12-27  USC00516128  0.14    71
19324    19324  19324 2016-12-28  USC00516128  0.14    71
19325    19325  19325 2016-12-29  USC00516128  1.03    69
19326    19326  19326 2016-12-30  USC00516128  2.37    65
19327    19327  19327 2016-12-31  USC00516128  0.90    65

[1517 rows x 6 columns]

In [15]:
# Identify the average temperature for June
avg_temp_june = june.tobs.mean()
june_temp = june["tobs"]
avg_temp_june

74.94411764705882

In [16]:
# Identify the average temperature for December
avg_temp_dec = december.tobs.mean()
december_temp = december["tobs"]
avg_temp_dec 

71.04152933421226

In [18]:
# Create collections of temperature data for June
june["station"].value_counts()

USC00513117    236
USC00519397    236
USC00519281    236
USC00519523    232
USC00516128    223
USC00514830    186
USC00511918    165
USC00517948    148
USC00518838     38
Name: station, dtype: int64

In [19]:
# Create collections of temperature data for december
december["station"].value_counts()

USC00519281    217
USC00513117    216
USC00519397    210
USC00519523    210
USC00516128    206
USC00514830    178
USC00511918    152
USC00517948    109
USC00518838     19
Name: station, dtype: int64

In [20]:
# Run paired t-test
stats.ttest_ind(june_temp, december_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis
The T-test result shows that there is difference in two datasets.
the pvalue is significant low (3.9), so there is significant
difference in temperature in June and December.
